File: app/src/main/java/rod/bailey/trafficatnsw/hazard/filter/IHazardFilter.java
package rod.bailey.trafficatnsw.hazard.filter;

import rod.bailey.trafficatnsw.json.hazard.XHazard;

public interface IHazardFilter {

	/**
	 * @param hazard
	 *            XHazard to be filtered. Must be non-null.
	 * @return true if teh given hazard is admitted (gets through) this filter
	 */
	public boolean admit(XHazard hazard);

}

File: app/src/main/java/rod/bailey/trafficatnsw/hazard/filter/IHazardFilter.kt
package rod.bailey.trafficatnsw.hazard.filter

import rod.bailey.trafficatnsw.hazard.data.XHazard

interface IHazardFilter {

	/**
	 * @param hazard XHazard to be filtered.
	 * @return true if the given hazard is admitted (gets through) this filter
	 */
	fun admit(hazard: XHazard): Boolean
}
